### PR TITLE
feature: update app to use v2 typings and DE APIs

### PR DESCRIPTION
**Pull request by zplata**
Looking to update this app to v2 Designer APIs - mostly the changes are in `insert-icon.ts`

### CODE REVIEW

File: src/insert-icon.ts [hunk 1]
```diff
@@ -1,38 +1,42 @@
...
+    if (attrKey === "class") {
       // skip adding class attributes because we don't have the css that goes with it
       return;
     }
     // set the attributes for the DOM element
-    domElement.setAttribute(attrKey, value);
+    attributePromises.push(newDomElement.setAttribute(attrKey, value));
   });
-  return domElement;
+  await Promise.all(attributePromises);
```
Comment (jamesmosier) [inline src/insert-icon.ts:20]: Perfect 🙌

**Review by jamesmosier — APPROVED** [review]
This look great 🎉  thank you so much for doing this! 

I did have one quick request...

Could you merge in `main` and then run `npm run format` and commit those updates? I just pushed up a `prettierrc` config file. I realized I never had this before, but in an effort to keep the diff of the upgrade between v1 and v2 the smallest we can I think it'd be worth avoiding any large formatting changes in this PR. Once you run `npm run format` you should see a smaller diff in this PR!

**Comment by zplata** [comment]
@jamesmosier yeah good call! just rebased with the new config, thanks for adding that

**Review by jamesmosier — APPROVED** [review]
thanks again 🎉 🚢

**Pull request merged**